rust/windows-client/src-tauri/src/debug_commands.rs

```diff
@@ -0,0 +1,185 @@
+//! CLI subcommands used to test features / dependencies before integrating
+//! them with the GUI, or to exercise features programmatically.
+
+use crate::prelude::*;
+use connlib_client_shared::{file_logger, Callbacks, Error, Session};
+use secrecy::SecretString;
+
+/// Test connlib and its callbacks.
+pub fn connlib(common_args: CommonArgs) -> Result<()> {
+    use connlib_client_shared::ResourceDescription;
+    use smbioslib::SMBiosSystemInformation as SysInfo;
+
+    #[derive(Clone)]
+    struct CallbackHandler {
+        handle: Option<file_logger::Handle>,
+    }
+
+    impl Callbacks for CallbackHandler {
+        type Error = std::convert::Infallible;
+
+        fn on_disconnect(&self, error: Option<&Error>) -> Result<(), Self::Error> {
+            tracing::error!("on_disconnect not implemented. Error: {error:?}");
+            Ok(())
+        }
+
+        fn on_error(&self, error: &Error) -> Result<(), Self::Error> {
+            tracing::error!("on_error not implemented. Error: {error}");
+            Ok(())
+        }
+
+        fn on_update_resources(
+            &self,
+            _resource_list: Vec<ResourceDescription>,
+        ) -> Result<(), Self::Error> {
+            tracing::error!("on_update_resources not implemented");
+            Ok(())
+        }
+
+        fn roll_log_file(&self) -> Option<PathBuf> {
+            self.handle
+                .as_ref()?
+                .roll_to_new_file()
+                .unwrap_or_else(|e| {
+                    tracing::debug!("Failed to roll over to new file: {e}");
+                    let _ = self.on_error(&Error::LogFileRollError(e));
+
+                    None
+                })
+        }
+    }
+
+    let (layer, handle) = file_logger::layer(std::path::Path::new("."));
+    setup_global_subscriber(layer);
+
+    // TODO: Is the SHA256 only intended to make the device ID fixed-length, or is it supposed to obfuscate the ID too? If so, we could add a pepper to defeat rainbow tables.
+
+    let data = smbioslib::table_load_from_device()?;
+    let device_id = if let Some(uuid) = data.find_map(|sys_info: SysInfo| sys_info.uuid()) {
+        tracing::info!("smbioslib got UUID");
+        uuid.to_string()
+    } else {
+        tracing::error!("smbioslib couldn't find UUID, making a random device ID");
+        uuid::Uuid::new_v4().to_string()
+    };
+
+    let mut session = Session::connect(
+        common_args.api_url,
+        SecretString::from(common_args.token),
+        device_id,
+        CallbackHandler {
+            handle: Some(handle),
+        },
+    )
+    .unwrap();
```

Comment:
For the GUI, yeah. This a debug command so the panic would already be shown in the terminal.
What if I put a TODO in gui.rs saying "fix or comment all unwraps before 1.0"? I need to do that anyway, since I was just unwrapping stuff without handling in many places